Stage 1 Frontend and API Submission

Megans-Tech-Life opened the pull request.

Megans-Tech-Life committed:
modals

Megans-Tech-Life committed:
second commit

Megans-Tech-Life committed:
project ready for first submit

Megans-Tech-Life committed:
fix github routing issue

Megans-Tech-Life committed:
fix router issue

Megans-Tech-Life committed:
fixed font errors

Megans-Tech-Life committed:
fix deploy error

Megans-Tech-Life committed:
fixed github deploy

Megans-Tech-Life committed:
fix import issue

Megans-Tech-Life committed:
typo

Megans-Tech-Life committed:
fixed image import issue

Megans-Tech-Life committed:
css edit

Megans-Tech-Life committed:
deploy works both locally and GitHub pages

Megans-Tech-Life committed:
fixed eslint errors

n817 commented:
Hello! I'm glad to say that you've done a really good job! Your project has been accepted.
I wish you success in the next stage and in your future projects!✨

Number of points: 89

# Summary:
### Performance Criteria

#### Markup and JSX
- [x] <!--7.4--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions. There is no horizontal scrolling on resolutions of `320px` and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--7.26--> Navigation between pages and links to external resources are working correctly. There are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and there are no hidden blocks.
- [x] <!--7.26--> Elements specified in the UI toolkit are used correctly (e.g., buttons, forms, text elements, etc).
- [x] <!--3.63--> Classes are named according to BEM specifications.
- [ ] <!--3.63--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.

* Please structure the HTML using semantic tags: use `<main>` for the main content.
The overall HTML structure should look like this:

`<header>...</header>`
`<main>`
  `about/home section`
`<main>`
`<footer>...</footer>`

- [x] <!--3.63--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--3.63--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--3.63--> Infrastructural project files are created using CRA.
- The markup has been converted into JSX:
  - [x] <!--3.63--> The markup is included within parentheses `( )`.
  - [x] <!--3.63--> The markup has been moved into the corresponding components.
- [ ] <!--3.63--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [ ] A `fonts` folder.

- [x] <!--3.63--> There are no linting errors or warnings.

#### React
- React components:
  - [x] <!--3.63--> Hooks are not used inside conditional statements or loops.
  - [x] <!--3.63--> Hooks are called in a component's main function.
  - [x] <!--3.63--> For class components, effects are described inside the component lifecycle methods.
- [x] <!--7.26--> Asynchronous API requests:
  - Requests can be made through the Fetch API or by using XMLHttpRequest. Third-party libraries (such as axios or jQuery) are not used.
  - API requests are contained in a separate file.
  - The chain for processing promises ends with a `catch()` block.
  - The first `then()` handler returns `res.json`.
- [x] <!--3.63--> Naming:
  - camelCase is used for function and variable names.
  - Only nouns are used as variable names.
  - Variable names clearly describe what is stored in them. If the project has several variables with similar data, then those variables have unique but descriptive names.
  - Descriptive names are used for functions, which reflect what they do.
  - Function names start with a verb.
  - JS classes and functional components are named using nouns and start with a capital letter.
  - Names must not include inappropriate or unclear abbreviations.
  - Сustom hook names start with `use`.
- [x] <!--3.63--> No third-party JavaScript libraries are used, unless absolutely necessary. If third-party libraries are connected, then they are used appropriately.

### Best Practices

#### Markup and JSX
- [x] <!--1.25--> The design reuses components wherever possible.
- [x] <!--1.25--> Fonts are connected using `@font-face`.
- [ ] <!--1.25--> The icons are exported in SVG format.

* All icons should be exported in SVG format.

- [x] <!--1.25--> Form elements should be highlighted when focused.
- [x] <!--1.25--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--1.25--> The use of `reset.css` is prohibited.

#### React
- [x] <!--1.25--> The initial state of state variables contain the correct data type.
- [x] <!--1.25--> API requests are described inside the `App` component.
- [x] <!--1.25--> There is no memory leak when hanging handlers. All handlers added with `addEventListener` are removed when the component is unmounted.
- [x] <!--1.25--> API error handling:
  - [x] The user receives a message in the case of an error.
- [x] <!--1.25--> Non-variable values (hard-coded constants) are named in all capital letters and stored in a separate configuration file.
- [ ] <!--1.25--> If you are using popups, then correct the opening and closing functionality is implemented. Popups can be closed using the cross button, overlay, or ESC key.

### Recommendations

#### Markup and JSX
- [x] <!--0.83--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--0.83--> Images have an `alt` attribute containing appropriate values.
- [ ] <!--0.83--> Raster and vector images have been optimized.

* It's better to reduce size of `Logo.png` image. Use the optimized images to save bandwidth and faster loading time of your website. You can use online-compression service, e.g. https://tinypng.com

#### React
- [x] <!--0.83--> For internal links in the application, components from the react-router library are used.
- [x] <!--0.83--> Semantically correct blocks are used for components. No `<div />` or other unnecessary HTML tags are used for components that consist of single-level blocks.
- [x] <!--0.85--> The code is clean and easy to understand:
  - The code is readable and clearly structured. Some parts of the code are explained with comments if needed.
  - There is no extra code: for example, when a variable is declared but not used, or there is some kind of redundant logic.
  - The code is formatted in the same way, and the indentation hierarchy is respected.

Megans-Tech-Life commented:
Excellent news thank you very much!

On Wed, Sep 17, 2025 at 5:05 PM Nikolai Gavriushin ***@***.***>
wrote:

> *n817* left a comment (mgracej91/Made_by_Mothers-frontend#1)
> <https://github.com/mgracej91/Made_by_Mothers-frontend/pull/1#issuecomment-3304561803>
>
> Hello! I'm glad to say that you've done a really good job! Your project
> has been accepted.
> I wish you success in the next stage and in your future projects!✨
>
> Number of points: 89
> Summary: Performance Criteria Markup and JSX
>
>    - The project is adapted for different screen resolutions and looks
>    correct on all intermediate resolutions. There is no horizontal scrolling
>    on resolutions of 320px and up. The scrollbar can't be hidden using
>    the overflow: hidden declaration.
>    - Navigation between pages and links to external resources are working
>    correctly. There are no broken links or links leading to an empty anchor on
>    the page, and external links open in a new tab. All project pages can be
>    accessed, and there are no hidden blocks.
>    - Elements specified in the UI toolkit are used correctly (e.g.,
>    buttons, forms, text elements, etc).
>    - Classes are named according to BEM specifications.
>    - Semantic HTML is used, meaning that semantic tags are used. All
>    elements are used correctly (e.g., a paragraph must be a paragraph, a list
>    must be a list). The DOM tree structure doesn't consist only of <div />
>    containers.
>
>
>    - Please structure the HTML using semantic tags: use <main> for the
>    main content.
>    The overall HTML structure should look like this:
>
> <header>...</header>
> <main>
> about/home section
> <main>
> <footer>...</footer>
>
>    -
>
>    The correct approach for positioning elements has been chosen and is
>    described using the correct syntax. For example, when positioning an
>    element absolutely, its parent block is relatively positioned.
>    -
>
>    flex or grid layouts are used to arrange elements whenever applicable.
>    -
>
>    Infrastructural project files are created using CRA.
>    -
>
>    The markup has been converted into JSX:
>    - The markup is included within parentheses ( ).
>       - The markup has been moved into the corresponding components.
>    -
>
>    Your project contains:
>    - An images
>       - A components folder with JS and CSS files for the components.
>       - A fonts folder.
>    -
>
>    There are no linting errors or warnings.
>
> React
>
>    - React components:
>       - Hooks are not used inside conditional statements or loops.
>       - Hooks are called in a component's main function.
>       - For class components, effects are described inside the component
>       lifecycle methods.
>    - Asynchronous API requests:
>       - Requests can be made through the Fetch API or by using
>       XMLHttpRequest. Third-party libraries (such as axios or jQuery) are not
>       used.
>       - API requests are contained in a separate file.
>       - The chain for processing promises ends with a catch() block.
>       - The first then() handler returns res.json.
>    - Naming:
>       - camelCase is used for function and variable names.
>       - Only nouns are used as variable names.
>       - Variable names clearly describe what is stored in them. If the
>       project has several variables with similar data, then those variables have
>       unique but descriptive names.
>       - Descriptive names are used for functions, which reflect what they
>       do.
>       - Function names start with a verb.
>       - JS classes and functional components are named using nouns and
>       start with a capital letter.
>       - Names must not include inappropriate or unclear abbreviations.
>       - Сustom hook names start with use.
>    - No third-party JavaScript libraries are used, unless absolutely
>    necessary. If third-party libraries are connected, then they are used
>    appropriately.
>
> Best Practices Markup and JSX
>
>    - The design reuses components wherever possible.
>    - Fonts are connected using @font-face.
>    - The icons are exported in SVG format.
>
>
>    - All icons should be exported in SVG format.
>
>
>    - Form elements should be highlighted when focused.
>    - The form must have placeholders, and there are required properties
>    for the fields.
>    - The use of reset.css is prohibited.
>
> React
>
>    - The initial state of state variables contain the correct data type.
>    - API requests are described inside the App component.
>    - There is no memory leak when hanging handlers. All handlers added
>    with addEventListener are removed when the component is unmounted.
>    - API error handling:
>       - The user receives a message in the case of an error.
>    - Non-variable values (hard-coded constants) are named in all capital
>    letters and stored in a separate configuration file.
>    - If you are using popups, then correct the opening and closing
>    functionality is implemented. Popups can be closed using the cross button,
>    overlay, or ESC key.
>
> Recommendations Markup and JSX
>
>    - System fonts are connected as alternatives to each of your fonts.
>    - Images have an alt attribute containing appropriate values.
>    - Raster and vector images have been optimized.
>
>
>    - It's better to reduce size of Logo.png image. Use the optimized
>    images to save bandwidth and faster loading time of your website. You can
>    use online-compression service, e.g. https://tinypng.com
>
> React
>
>    - For internal links in the application, components from the
>    react-router library are used.
>    - Semantically correct blocks are used for components. No <div /> or
>    other unnecessary HTML tags are used for components that consist of
>    single-level blocks.
>    - The code is clean and easy to understand:
>       - The code is readable and clearly structured. Some parts of the
>       code are explained with comments if needed.
>       - There is no extra code: for example, when a variable is declared
>       but not used, or there is some kind of redundant logic.
>       - The code is formatted in the same way, and the indentation
>       hierarchy is respected.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mgracej91/Made_by_Mothers-frontend/pull/1#issuecomment-3304561803>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BPLNPGT3XQZMVWTDOGRULR33THEKZAVCNFSM6AAAAACGWOZY4GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTGMBUGU3DCOBQGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Megans-Tech-Life committed:
Update frontend repo link in README

Megans-Tech-Life committed:
Update frontend repo link in README

Megans-Tech-Life committed:
Revise README for clarity and detail

Updated README to enhance project description, features, and setup instructions.

Megans-Tech-Life committed:
update
